tmt/steps/provision/__init__.py

```diff
@@ -1646,6 +1663,60 @@ def essential_requires(cls) -> list['tmt.base.Dependency']:
 
         return []
 
+    def acquire_log(self, logname: str) -> Optional[str]:
+        """
+        Fetch and return content of a log.
+
+        :param logname: name of the log.
+        :returns: content of the log, or ``None`` if the log cannot be retrieved.
+        """
+        raise NotImplementedError
+
+    def store_log(self, path: Path, content: str, logname: Optional[str] = None) -> None:
+        """
+        Save log content to a file.
+
+        :param path: a path to save into, could be a directory
+            or a file path.
+        :param content: content of the log.
+        :param logname: name of the log, if not set, logpath
+            is supposed to be a file path.
+        """
+        # if path is file path
+        if not path.is_dir():
+            path.write_text(content)
+        # if path is a directory
+        elif logname:
+            (path / logname).write_text(content)
+        else:
+            raise tmt.utils.GeneralError('Log path is a directory but log name is not defined.')
+
+    def fetch_logs(
+        self, dirpath: Optional[Path] = None, guest_logs: Optional[list[GuestLog]] = None
+    ) -> None:
+        """
+        Get log content and save it to a directory.
+
+        :param dirpath: a directory to save into. If not set, step's working directory
+            (:py:attr:`workdir`) or current working directory will be used.
+        :param lognames: name list of logs need to be handled. If not set, all guest logs
+            would be collected, as reported by :py:attr:`lognames`.
+        """
+
+        guest_logs = guest_logs or self.guest_logs or []
+
+        dirpath = dirpath or (self.workdir / 'logs' if self.workdir else None) or Path.cwd()
```

Comment:
There should be a `logdir` property of `Guest` class that does this: every part of code interested in finding where guest's logs are should use `guest.logdir` instead of constructing the part on their own. Then you should be able to simply run `guest.logdir.mkdir(parents=True, exists_ok=True)` instead of checking `self.workdir` again below.